hail/src/main/scala/is/hail/HailContext.scala

```diff
@@ -124,7 +125,9 @@ object HailContext {
     info(s"Running Hail version ${ theContext.version }")
 
     // needs to be after `theContext` is set, since this creates broadcasts
-    backend.addDefaultReferences()
+    if (addDefaultReferences) {
+      backend.addDefaultReferences()
+    }
```

Comment:
I think you can get rid of this entirely and just let the backends always add the default references. ReferenceGenome use lazy val for their broadcasted values. I don't see why we would start a Backend without immediately starting a HailContext.